### PR TITLE
Created PUT '/lesson'

### LINKED ISSUE #81: CRUD 'language/lesson'
**Is your feature request related to a problem? Please describe.**
We need endpoints for creating, reading, updating, and deleting lessons.

**Describe the solution you'd like**
The endpoints can be: 
- POST 'language/lesson' for creating a new lesson.
- GET 'language/lesson' for getting lesson data and all of the vocab that belong to it.
- PUT 'language/lesson' for updating a lesson's name, description, and/or order.
- DELETE 'language/lesson' for deleting a lesson

**Make sure you understand what [lesson data we need to store in MongoDB.](https://github.com/hack4impact-uiuc/7000-languages/blob/dev/api/src/models/language.js). And, don't forget to write tests/consider edge cases!**

Take a look at the [issues for `CRUD 'language/courses'`](https://github.com/hack4impact-uiuc/7000-languages/issues/7) for more ideas, hints, and code.

**Resources**
Take a look at the [Mongoose documentation](https://mongoosejs.com/docs/models.html) for more clarification.

### PR DESCRIPTION
**Pull request by ashayp22**
## Status:

🚀 

## Description

- Wrote an endpoint for making changes to more than one lesson at once. This will be useful for the Manage Units/Lessons page, where we unselect/select and re-order lessons and units.
- Wrote Mongoose validation that checks if a lesson has a unique `_order` before saving.
- Remove unnecessary code from previously created endpoints and updated their accompanying tests.

Fixes #81

### CODE REVIEW

File: api/src/models/language.js [hunk 1]
```diff
@@ -53,8 +55,55 @@ const Lesson = new mongoose.Schema({
 
 Lesson.index({ _course_id: 1, _unit_id: 1, _order: 1 });
 
+/* Validation Methods */
+
```
Comment (ashayp22) [inline api/src/models/language.js:59]: get rid of TODO

File: api/src/models/language.js [hunk 2]
```diff
@@ -53,8 +55,55 @@ const Lesson = new mongoose.Schema({
...
+// TODO: update find to consider course and unit id
+
+/**
+ * Checks if the step number is unique among all other steps. This
```
Comment (ashayp22) [inline api/src/models/language.js:63]: order

File: api/src/models/language.js [hunk 3]
```diff
@@ -53,8 +55,55 @@ const Lesson = new mongoose.Schema({
...
+ * function is meant to be called in transactions.
+ * @param {Number} stepNumber The step number to check.
+ * @param {String} stepKey The stepKey for the step we're validating.
```
Comment (ashayp22) [inline api/src/models/language.js:66]: replace with proper documentation

File: api/src/models/language.js [hunk 4]
```diff
@@ -53,8 +55,55 @@ const Lesson = new mongoose.Schema({
...
+ * @param {Object} session The session object for this transaction.
+ * @returns True if the step number is unique.
+ */
+const isUniqueOrder = async (params, _id, model, session = null) => {
+  // Find all steps with this step number
```
Comment (ashayp22) [inline api/src/models/language.js:71]: replkace with proper documentation

**Review by ashayp22 — COMMENTED** [review]
glhf
